app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java

```diff
@@ -673,21 +685,22 @@ public boolean onBackPressed() {
     // Save the file
     // Only supports java.io.File. TODO: Android Content
     public boolean saveDocument(boolean forceSaveEmpty) {
-        if (isAdded() && _hlEditor != null && _hlEditor.getText() != null) {
+        if (_isTextChanged && isAdded() && _hlEditor != null && _hlEditor.getText() != null) {
```

Comment:
Only writing if `_isTextChanged == true`. Should address several cases from #1303 etc